data/test/scenarios/wml_tests/ScenarioWML/EventWML/events-test_filters.cfg

```diff
@@ -294,20 +294,175 @@
+            [/attack]
+        [/do_command]
+        [end_turn][/end_turn]
+    [/event]
+    [event]
+        name=attack
+        first_time_only=no
+        [filter_attack]
+            special_id_active=ability_poison_blade
+        [/filter_attack]
+        {ASSERT ({VARIABLE_CONDITIONAL side_number equals 2})}
+        {VARIABLE_OP triggers_on_attack add 1}
+    [/event]
+    [event]
+        name=attack
+        first_time_only=no
+        [filter_second_attack]
+            special_id_active=ability_poison_blade
+        [/filter_second_attack]
+        {ASSERT ({VARIABLE_CONDITIONAL side_number equals 1})}
+        {VARIABLE_OP triggers_on_defense add 1}
+    [/event]
+#enddef
+
 #####
 # API(s) being tested: [event][filter_attack],[event][filter_second_attack]
 ##
 # Actions:
 # Define events filtering based on weapon specials.
 # Give Bob a poison special iff the opponent has a specific drain special.
-# Give Alice a drain special iff the opponent has a specific poison special.
+# Give Alice that drain special iff the opponent has a blade attack.
 # Have side 1's unit attack side 2's unit.
 # Have side 2's unit attack side 1's unit.
 ##
 # Expected end state:
 # Both events that should trigger for Bob do trigger.
 #####
 {GENERIC_UNIT_TEST event_test_filter_attack_opponent_weapon_condition (
+    {SPECIAL_CONDITION (type=blade)}
+    [event]
+        name=turn 2
+        {ASSERT ({VARIABLE_CONDITIONAL triggers_on_attack equals 1})}
+        {ASSERT ({VARIABLE_CONDITIONAL triggers_on_defense equals 1})}
+        {SUCCEED}
+    [/event]
+)}
+
+#####
+# API(s) being tested: [event][filter_attack],[event][filter_second_attack]
+##
+# Actions:
+# Define events filtering based on absence of weapon specials.
+# Give Bob a poison special iff the opponent has a specific drain special, and that special is active.
+# Give Alice a drain special iff the opponent has a specific poison special, and that special is active.
+# Have side 1's unit attack side 2's unit.
+# Have side 2's unit attack side 1's unit.
+##
+# Expected end state:
+# BROKE STRICT because testing the special_id_active conditions led to infinite recursion.
+# The events don't trigger. Both specials depend on the other one being active, so the calculation would need infinate recursion; instead, the engine decides that both are inactive.
+#####
+{GENERIC_UNIT_TEST event_test_filter_attack_opponent_weapon_condition_no_triggered (
+    {SPECIAL_CONDITION (special_id_active=ability_poison_blade)}
+    [event]
+        name=turn 2
+        {ASSERT ({VARIABLE_CONDITIONAL triggers_on_attack equals 0})}
+        {ASSERT ({VARIABLE_CONDITIONAL triggers_on_defense equals 0})}
+        {SUCCEED}
+    [/event]
+)}
+
+#undef SPECIAL_CONDITION
+
+#####
+# API(s) being tested: [event][filter_attack],[event][filter_second_attack]
+##
+# Actions:
+# Define specials filtering based on weapon specials.
+# Give Bob a poison special iff own a same special active or a blade or pierce attack.
```

Comment:
Not sure what is meant by 'a same special active' for both Bob here and Alice below.

Reply:
if special check itself.